game-headed/src/main/java/org/triplea/game/client/ui/javafx/screens/MapSelection.java

```diff
@@ -0,0 +1,149 @@
+package org.triplea.game.client.ui.javafx.screens;
+
+import com.google.common.annotations.VisibleForTesting;
+import games.strategy.engine.framework.ui.GameChooserEntry;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+import javafx.application.Platform;
+import javafx.fxml.FXML;
+import javafx.geometry.Pos;
+import javafx.scene.Node;
+import javafx.scene.control.Button;
+import javafx.scene.control.ContentDisplay;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.TilePane;
+import javafx.scene.web.WebView;
+import lombok.AccessLevel;
+import lombok.Setter;
+import org.triplea.game.client.parser.DefaultGameDetector;
+import org.triplea.game.client.parser.GameDetector;
+import org.triplea.game.client.ui.javafx.screen.ControlledScreen;
+import org.triplea.game.client.ui.javafx.screen.ScreenController;
+import org.triplea.game.client.ui.javafx.util.FxmlManager;
+import org.triplea.util.LocalizeHtml;
+
+public class MapSelection implements ControlledScreen<ScreenController<FxmlManager>> {
+
+  private final GameDetector gameDetector;
+  private final BiFunction<String, String, String> linkLocalizer;
+
+  @Setter(
+      value = AccessLevel.PACKAGE,
+      onMethod_ = {@VisibleForTesting})
+  @FXML
+  private Node root;
+
+  @FXML private TilePane mapContainer;
+
+  @Setter(
+      value = AccessLevel.PACKAGE,
+      onMethod_ = {@VisibleForTesting})
+  @FXML
+  private Node previewContainer;
+
+  @Setter(
+      value = AccessLevel.PACKAGE,
+      onMethod_ = {@VisibleForTesting})
+  @FXML
+  private WebView previewWindow;
+
+  @FXML private Button selectButton;
+  @FXML private Button detailsButton;
+
+  @Setter(
+      value = AccessLevel.PACKAGE,
+      onMethod_ = {@VisibleForTesting})
+  private GameChooserEntry selectedGame;
+
+  private boolean loaded = false;
+
+  private ScreenController<FxmlManager> screenController;
+
+  // Constructor used via FXML reflection
+  // to initialize this controller.
+  @SuppressWarnings("unused")
+  public MapSelection() {
+    this(new DefaultGameDetector(), LocalizeHtml::localizeImgLinksInHtml);
+  }
+
+  @VisibleForTesting
+  MapSelection(
+      final GameDetector gameDetector, final BiFunction<String, String, String> linkLocalizer) {
+    this.gameDetector = gameDetector;
+    this.linkLocalizer = linkLocalizer;
+  }
+
+  private Node createMapListing(final GameChooserEntry gameChooserEntry) {
+    final var button = new Button(gameChooserEntry.getGameName());
+    // Placeholder image
+    final var imageView =
+        new ImageView(new Image("https://triplea-game.org/images/missing_map.png", true));
```

Comment:
- What happens when the map download screen is opened without a network connection?
- Should 'missing_map.png' be hosted on triplea-game.org? Would it be better to bundle the missing_map.png with the game so that a network fetch is not needed? Any concern if 'missing_map.png' were moved and we did not realize game engines needed it? Is there also risk that we repeat this pattern for other images, how do we keep track of which images are hosted on website and used by game engines vs those that are used only by website?

Reply:
1. Good question. I would assume nothing (except for a stacktrace in the console) and a missing image, because I tried putting an svg there which isn't supported by JavaFX which did exactly this (without the stacktrace though). I'd have to test this though.
2. It's probably better to bundle it with the game. Initially I wasn't sure if I wanted to implement "real" thumbnails as part of this PR, which is why I used some random placeholder image for this.

Reply:
I just checked using an invalid domain, just like I expected nothing happens.
The button stays blank, and not even an exception is logged